errorkind/errorkind_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019, OpenTelemetry Authors
```

Comment:
I am worried about the proliferation of top-level packages. Is there a better place for this package, somewhere not top-level?
If this is for receivers only then it can go to `receivers` package.
Or maybe have an `errors` package and place this and `commonerrors.go` in it (if this is supposed to be used by other parts of the code)?

Reply:
I agree with the concern. This shouldn't be only for `receivers`.

I don't like `errors` because it is a standard package, but moving it to errors/errorkind instead of top level, this way if more error related stuff comes up we put it under the same path

Reply:
Done